### PR TITLE
Remove more instances of `UNKNOWN_NULL_COUNT`

### PR DESCRIPTION
**Pull request by vyasr**
## Description

Contributes to #11968.

## Checklist
- [x] I am familiar with the [Contributing Guidelines](https://github.com/rapidsai/cudf/blob/HEAD/CONTRIBUTING.md).
- [x] New or existing tests cover these changes.
- [x] The documentation is up to date with these changes.

### CODE REVIEW

**Review by jlowe — APPROVED** [review]

File: cpp/examples/basic/CMakeLists.txt [hunk 1]
```diff
@@ -1,4 +1,4 @@
-# Copyright (c) 2020-2022, NVIDIA CORPORATION.
+# Copyright (c) 2020-2023, NVIDIA CORPORATION.
```
Comment (ttnghia) [inline cpp/examples/basic/CMakeLists.txt:1]: Merge conflict? Nothing changed in this (and below) files.

Comment (vyasr) [reply cpp/examples/basic/CMakeLists.txt:1]: [Commented on this above](https://github.com/rapidsai/cudf/pull/13134#issuecomment-1507381612)

File: cpp/src/io/utilities/column_buffer.cpp [hunk 2]
```diff
@@ -116,19 +116,17 @@ std::unique_ptr<column> make_column(column_buffer& buffer,
...
         auto data      = contents.data.release();
         auto null_mask = contents.null_mask.release();
 
-        auto uint8_col = std::make_unique<column>(data_type{type_id::UINT8},
-                                                  data->size(),
-                                                  std::move(*data),
-                                                  std::move(*null_mask),
-                                                  UNKNOWN_NULL_COUNT);
+        auto uint8_col = std::make_unique<column>(
+          data_type{type_id::UINT8}, data->size(), std::move(*data), std::move(*null_mask), 0);
```
[the remark this answers is not in the document]
Comment (vyasr) [reply cpp/src/io/utilities/column_buffer.cpp:129]: Yeah I'm fine switching to just constructing an empty buffer.

**Review by ttnghia — APPROVED** [review]

**Review by davidwendt — APPROVED** [review]

**Review by bdice — APPROVED** [review]
CMake LGTM.

**Pull request merged**